File: Python/kenkensolver.py
#!/usr/bin/env python

from board import Board
from group import Group
from operation import Operation

size = 4
groups = {
	"a": Group(6, Operation.get("+")),
	"b": Group(7, Operation.get("+")),
	"c": Group(4, Operation.get("+")),
	"d": Group(7, Operation.get("+")),
	"e": Group(7, Operation.get("+")),
	"f": Group(9, Operation.get("+"))
}
#acee
#adef
#addf
#bbbf
pattern = 'aceeadefaddfbbbf'

board = Board(size)

for i in range(0, size * size):
	groupName = pattern[i]
	group = groups[groupName]
	cell = board.cells[i]
	group.add(cell)

board.groups = groups.values()

board.solve()

File: Ruby/kenkensolver.rb
#!/usr/bin/env ruby

require './board'
require './group'
require './operation'

size = 3
groups = {
	"a": Group.new(3, Operation.+),
	"b": Group.new(5, Operation.+),
	"c": Group.new(3, Operation.+),
	"d": Group.new(4, Operation.+),
	"e": Group.new(3, Operation.+)
}
#aad
#bbd
#cee
pattern = "aadbbdcee"

board = Board.new(size)

for i in 0...size*size
	groupName = pattern[i].to_sym
	group = groups[groupName]
	cell = board.cells[i]
	group.cells << cell
end

board.groups = groups.values

board.solve()